apps/web-giddh/src/app/project-wise-accounting/list/project-wise-accounting.component.ts

```diff
@@ -0,0 +1,391 @@
+    /** ReplaySubject to handle component's lifecycle */
+    private destroyed$: ReplaySubject<boolean> = new ReplaySubject(1);
+    /** If true, the project search box is displayed */
+    public showProductSearch: boolean = false;
+    /** This will use for displayed table columns */
+    public displayedColumns: string[] = ['Sno', 'Name', 'Archive Status', 'Status', 'Symbol', 'Action'];
+    /** This will store selected date range to use in api */
+    public selectedDateRange: any;
+    /** This will store selected date range to show on UI */
+    public selectedDateRangeUi: any;
+    /** This will store universalDate */
+    public universalDate: any;
+    /** Selected from date */
+    public fromDate: string;
+    /** Selected to date */
+    public toDate: string;
+    /** Selected range label */
+    public selectedRangeLabel: any = "";
+    /** This will store modal reference */
+    public modalRef: BsModalRef;
+    /** This will store the x/y position of the field to show datepicker under it */
+    public dateFieldPosition: any = { x: 0, y: 0 };
+    /** This will store available date ranges */
+    public datePickerOption: any = GIDDH_DATE_RANGE_PICKER_RANGES;
+    /** Getter for the 'name' form control from the companyListForm. */
+    public get projectName(): FormControl {
+        return this.companyListForm.get('projectName') as FormControl
+    }
+
+
+    constructor(
+        public dialog: MatDialog,
+        private componentStore: ProjectWiseAccountingComponentStore,
+        private generalService: GeneralService,
+        private fb: FormBuilder,
+        private changeDetection: ChangeDetectorRef,
+        private modalService: BsModalService
+    ) {
+        this.componentStore.activeCompany$.pipe(takeUntil(this.destroyed$)).subscribe(activeCompany => {
+            if (activeCompany) {
+                this.activeCompany = activeCompany;
+                this.setDefaultProject();
+            }
+        });
+    }
+
+    /**
+     * Lifecycle hook for component initialization.
+     *
+     * @memberof ProjectWiseAccountingListComponent
+     */
+    public ngOnInit(): void {
+        this.initCompanyListForm();
+        this.getAllProjectList();
+        this.companyListForm.get('projectName').valueChanges.pipe(debounceTime(700), takeUntil(this.destroyed$)).subscribe((searchedText) => {
+            if (searchedText) {
+                this.projectListRequest.q = searchedText;
+                this.getAllProjectList();
+            }
+        });
+
+        this.componentStore.projectsList$.pipe(takeUntil(this.destroyed$)).subscribe(projectList => {
+            if (projectList) {
+                this.totalResults = projectList.totalItems;
+                this.dataSource = this.addProfitAndLossKey(projectList.results);
+            }
+        });
+
+        this.componentStore.projectProfitDetails$.pipe(takeUntil(this.destroyed$)).subscribe(profitandloss => {
```

Comment:
```suggestion
        this.componentStore.projectProfitDetails$.pipe(takeUntil(this.destroyed$)).subscribe(response => {
```

Correct in over all observable